tests/test_cache.py

```diff
@@ -0,0 +1,238 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import random
+
+import pytest
+import torch
+
+from tests.ops.cache_op import reshape_and_cache, reshape_and_cache_flash
+from tests.utils import (create_kv_caches_with_random,
+                         create_kv_caches_with_random_flash, opcheck)
+
+DTYPES = [torch.half, torch.bfloat16]
+NUM_TOKENS = [42]  # Arbitrary values for testing
+NUM_HEADS = [8]  # Arbitrary values for testing
+NUM_LAYERS = [1]
+HEAD_SIZES = [64, 80, 120, 256]
+BLOCK_SIZES = [8, 16, 32]
+NUM_BLOCKS = [1024]  # don't make it too large. e.g. [36000] will OOM
+SEEDS = [0]
+DEVICES = [
+    f"xpu:{i}" for i in range(1 if torch.xpu.device_count() == 1 else 2)
+]
+KV_CACHE_DTYPE = ["auto"]
```

Comment:
we didn't add fp8 here due to acc issue right? can we add some todo or FIXME here?